sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationClient.java

```diff
@@ -295,30 +297,34 @@ public ConfigurationSetting updateSetting(ConfigurationSetting setting) {
      *
      * <p>Retrieve the setting with the key "prodDBConnection".</p>
      *
-     * {@codesnippet com.azure.data.applicationconfig.configurationclient.getSetting#string}
+     * {@codesnippet com.azure.data.applicationconfig.configurationclient.getSetting#string-string-OffsetDateTime}
      *
      * @param key The key of the setting to retrieve.
+     * @param label Optional, the label of the setting to retrieve.
+     * @param acceptDateTime Optional, to access a past state of the configuration setting.
      * @return The {@link ConfigurationSetting} stored in the service, or {@code null}, if the configuration value does
      * not exist or the key is an invalid value (which will also throw ServiceRequestException described below).
      * @throws IllegalArgumentException If {@code key} is {@code null}.
      * @throws ResourceNotFoundException If a ConfigurationSetting with {@code key} does not exist.
      * @throws HttpResponseException If {@code key} is an empty string.
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
-    public ConfigurationSetting getSetting(String key) {
-        return getSetting(new ConfigurationSetting().setKey(key), Context.NONE).getValue();
+    public ConfigurationSetting getSetting(String key, String label, OffsetDateTime acceptDateTime) {
```

Comment:
Follow-up with the group on whether we want the DateTime overload to be the simple overload. The date time use case is a bit more advanced.

Additionally, we need to have this parameter in the maximal overload. It should allow the same functionality as this one.

Reply:
@JonathanGiles  Mind giving some guidance here, having the DateTime in the maximal overload would diverge from what the other languages have.

Reply:
I don't mind have multiple 'maximal' overloads. To me, it seems like having one that takes (key, label) is a minimal and (key, label, asOf) is a local maximal.